dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -148,6 +183,44 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     }
     CATCH_RETURN()
 
+    HRESULT __stdcall PushNotificationChannel::OnRawNotificationReceived(unsigned int payloadLength, _In_ byte* payload, _In_ HSTRING /*correlationVector */) noexcept
+    {
+        try
+        {
+            BOOL foregroundHandled = false;
+            THROW_IF_FAILED(InvokeAll(payloadLength, payload, &foregroundHandled));
+            THROW_HR(E_INVALIDARG);
+        }
+        catch(...)
+        {
+            //TODO: Capture the HRESULT in telemetry
+            winrt::com_array<uint8_t> payloadArray{ payload, payload + (payloadLength * sizeof(uint8_t)) };
+            std::string commandLine = "----WindowsAppRuntimePushServer:-Payload:\"";
+            commandLine.append(reinterpret_cast<char*>(payload), payloadLength);
+            commandLine.append("\"");
+
+            wil::unique_cotaskmem_string processName;
+            THROW_IF_FAILED(GetCurrentProcessPath(processName));
+
+            const std::string processNameAsUtf8String = ConvertProcessNameToUtf8String(processName.get());
```

Comment:
Did you look into @danielayala94 's code change for special characters and protocol launch? I think we are introducing the same bug here which he fixed. Hence its important to share code through utility helpers so that we don't redo the same mistakes.

Reply:
Agreed. I totally missed that. I have integrated that change into my code and validated it. I have created a helper for the same. Thanks.